tests/langchain/test_langchain_model_export.py

```diff
@@ -736,3 +739,29 @@ def test_unsupported_class():
     ):
         with mlflow.start_run():
             mlflow.langchain.log_model(llm, "fake_llm")
+
+
+def test_agent_with_unpicklable_tools():
+    _, tmp_path = tempfile.mkstemp()
+    with open(tmp_path, mode="w") as temp_file:
+        # files that aren't opened for reading cannot be pickled
```

Comment:
/ai rewrite this code using `tmp_path` fixture

Reply:
ohhh that's what you meant, that's pretty cool!

Reply:
Looks good, thanks ai!

Reply:
@daniellok-db Can we apply the suggestion above?